server/controllers/booking.js

```diff
@@ -0,0 +1,67 @@
+const Booking = require("../models/Booking");
+const Profile = require("../models/Profile");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+
+// @route GET /bookings
+// @desc list of bookings
+// @access Private
+exports.getBookings = asyncHandler(async (req, res, next) => {
+  const profile = await Profile.findOne({ userId: req.user.id });
+  let bookings;
+  if (profile.accountType === "pet_owner") {
+    bookings = await Booking.find({ userId: req.user.id });
+  } else {
+    bookings = await Booking.find({ sitterId: req.user.id });
+  }
+
+  const bookingInfos = [];
+  for (let booking of bookings) {
+    booking = booking.toObject();
+    booking.petOwner = await User.findById(booking.userId);
+    booking.sitter = await User.findById(booking.sitterId);
+    bookingInfos.push(booking);
+  }
```

Comment:
Instead of this extra query, you could chain mongoose `populate` to the `findById` query to pull these fields from the DB on lines 13 and 15.